### ISSUE TITLE
Invalidate cache when ASF-ui version has changed

### ISSUE THREAD

**Issue by MrBurrBurr** [JustArchiNET/ASF-ui#1330]
https://github.com/JustArchiNET/ASF-ui/issues/1325#issuecomment-797053433

**Comment by JustArchi** [comment]
Leaving a note that it requires totally different approach, as was discussed on our discord: https://ptb.discord.com/channels/267292556709068800/332735075315744768/827505253607407626

**Comment by Aareksio** [comment]
TL;DR on the above - use bundle hashes as any sane project does for busting cache

How to do this, the most sane approach I thought of so far:
- use [html-webpack-plugin](https://webpack.js.org/plugins/html-webpack-plugin/) to generate index.html with bundle included
- feed custom index.html template to the above plugin, include inline script in the template
- the inline script shall detect base path similarly to how it is [already done](https://github.com/JustArchiNET/ASF-ui/blob/main/src/static/index.html#L16-L39)
- the inline script shall read the main bundle path from the script tag and replace it accordingly to real base path
- remember to also set `window.__BASE_PATH__`, as it is [used by webpack for loading other resources](https://github.com/JustArchiNET/ASF-ui/blob/main/src/public-path.js)
- we could optimize for the seemingly most common entry point, which is ASF's Kestrel running on localhost

**Comment by MrBurrBurr** [comment]
I believe we can extend this issue with the following:

When we detect that an ASF-ui update has happened (compare saved asf-ui commit hash with current one) we should delete certain properties from local storage. Those properties should be:
- `asf-ui:cache:releases`
- `asf-ui:cache:asf-commands:*`
- `asf-ui:cache:parameter-descriptions:*` 

The reason behind this is that we do not want to wait for the cache to be invalid but instead want to force a reload since it is almost certain that some content of one of those properties has changed.

**Comment by JustArchi** [comment]
How does that fix loading half of the ASF-ui with the user unable to navigate through the program before he invalidates entire cache?

Having older version of releases, asf commands or parameter descriptions doesn't sound like a root cause for being unable to navigate to bots window or any other tab.

**Comment by Botan626** [comment]
I think now @MrBurrBurr wants to invalidate cache **and** delete some properties from local storage at the same time.

**Comment by JustArchi** [comment]
Then it's alright thing to do, but I don't see how it's necessary to do it, user installing new version doesn't affect the releases I've published on GitHub, commands on the wiki, or "parameter descriptions", whatever that is, probably info from configuration page of the wiki. That should follow existing procedure of being cached for some time (like 6h, maybe up to 24h) and independent of updates.

**Comment by Botan626** [comment]
But just recently I couldn't find `points` command in the commands list in ASF-ui after I updated to newer version of ASF.
So user updates to newer version of ASF via ASF-ui, sees there's a new command added, he wants to read description about it in ASF-ui and he can't, he has to wait for 6 hours max, but most likely he wouldn't know that after 6 hours max some properties in local storage will be updated and then he will have a chance to do it.

**Comment by JustArchi** [comment]
You could also skip the update and have it cached before you updated to it. Unless ASF-ui actually fetches wiki page from release date, then yes, clearing it is a good idea.
